src/FondOfSpryker/Zed/CompanyUserCompanyAssigner/Business/Model/CompanyRole.php

```diff
@@ -0,0 +1,84 @@
+     */
+    protected $companyFacade;
+
+    /**
+     * @var \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Dependency\Facade\CompanyUserCompanyAssignerToCompanyRoleFacadeInterface
+     */
+    protected $companyRoleFacade;
+
+    /**
+     * @var \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Dependency\Facade\CompanyUserCompanyAssignerToCompanyTypeFacadeInterface
+     */
+    protected $companyTypeFacade;
+
+    /**
+     * @var \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Business\Model\CompanyUser
+     */
+    protected $companyUser;
+
+    /**
+     * @var \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Dependency\Facade\CompanyUserCompanyAssignerToCompanyUserFacadeInterface
+     */
+    protected $companyUserFacade;
+
+    /**
+     * @param \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Dependency\Facade\CompanyUserCompanyAssignerToCompanyFacadeInterface $companyFacade
+     * @param \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Dependency\Facade\CompanyUserCompanyAssignerToCompanyRoleFacadeInterface $companyRoleFacade
+     * @param \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Dependency\Facade\CompanyUserCompanyAssignerToCompanyTypeFacadeInterface $companyTypeFacade
+     * @param \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Business\Model\CompanyUser $companyUser
+     * @param \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Dependency\Facade\CompanyUserCompanyAssignerToCompanyUserFacadeInterface $companyUserFacade
+     */
+    public function __construct(
+        CompanyUserCompanyAssignerToCompanyFacadeInterface $companyFacade,
+        CompanyUserCompanyAssignerToCompanyRoleFacadeInterface $companyRoleFacade,
+        CompanyUserCompanyAssignerToCompanyTypeFacadeInterface $companyTypeFacade,
+        CompanyUser $companyUser,
+        CompanyUserCompanyAssignerToCompanyUserFacadeInterface $companyUserFacade
+    ) {
+        $this->companyFacade = $companyFacade;
+        $this->companyRoleFacade = $companyRoleFacade;
+        $this->companyTypeFacade = $companyTypeFacade;
+        $this->companyUser = $companyUser;
+        $this->companyUserFacade = $companyUserFacade;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CompanyUserTransfer $companyUserTransfer
+     *
+     * @return \Generated\Shared\Transfer\CompanyRoleCollectionTransfer
+     */
+    public function getCompanyUserRoleCollection(
+        CompanyUserTransfer $companyUserTransfer
+    ): CompanyRoleCollectionTransfer {
+         return $this->companyRoleFacade
+             ->getCompanyRoleCollection(
+                 (new CompanyRoleCriteriaFilterTransfer())
+                     ->setIdCompanyUser($companyUserTransfer->getIdCompanyUser()),
+             );
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CompanyUserTransfer $manufacturerCompanyUserTransfer
+     *
+     * @return void
+     */
+    public function updateNonManufacturerUsersCompanyRole(CompanyUserTransfer $manufacturerCompanyUserTransfer): void
+    {
+    }
```

Comment:
Why is this function empty? Should this throw an Exception maybe that it hasn't been implemented yet?